src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/transport/OpenDistroSSLDualModeConfig.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Portions Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.security.ssl.transport;
+
+import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+
+public class OpenDistroSSLDualModeConfig {
+
+    public static final Setting<Boolean> SSL_DUAL_MODE_SETTING = Setting.boolSetting(ConfigConstants.OPENDISTRO_SECURITY_SSL_DUAL_MODE_ENABLED,
+            true, Setting.Property.NodeScope, Setting.Property.Dynamic, Setting.Property.Filtered);
+
+    private static final Logger logger = LogManager.getLogger(OpenDistroSSLDualModeConfig.class);
+
+    private static OpenDistroSSLDualModeConfig INSTANCE;
```

Comment:
IMO, this class should not be a singleton.

Reply:
We determine the SSL_DUAL_MODE setting value based on a combination of the value from elasticsearch.yml file and corresponding cluster setting value. The object provides access to the value of SSL Dual mode config, which once set can only be modified via the cluster setting. The object is used during the setup of new connections in the transport server and transport client code to check the dual mode setting.
For this use case the class was designed as a singleton since we want to object to be created only once and used multiple times. If class is not singleton we would need to pass the created object to all places it is being used in.

Reply:
Create instance in the plugin class and pass it around if necessary.

Reply:
changing this class to be not a singleton, instead creating the object in plugin class and passing it to others